### PR TITLE
numbering issues with slash

### PR DESCRIPTION
**Pull request by socheres**
https://github.com/ubtue/DatenProbleme/issues/1370
https://github.com/ubtue/DatenProbleme/issues/1355

### CODE REVIEW

File: ubtue_Brill.js [hunk 1]
```diff
@@ -83,10 +83,33 @@ function postProcess(doc, item) {
...
             item.itemType = "journalArticle";
 }
 
+function extractErscheingunssjahr(date) {
+	let publicationYear = date.trim().match(/\d{4}/)[0];
+	return publicationYear;
+}
+
+function extractBerichtsjahr(dateEntry) {
+	let extractDate = dateEntry.split('(')[1].split(')')[0];
```
Comment (jriedl) [inline ubtue_Brill.js:109]: Wouldn't using a regex be more appropriate? With `split` the expression fails if there are no brackets.

Comment (socheres) [reply ubtue_Brill.js]: I changed the code, but i am not sure if using the match method with regex does not cause an error as well.

File: ubtue_Brill.js [hunk 2]
```diff
@@ -106,8 +106,8 @@ function extractErscheingunssjahr(date) {
 }
 
 function extractBerichtsjahr(dateEntry) {
-	let extractDate = dateEntry.split('(')[1].split(')')[0];
-		return extractDate;
+	let extractDate = dateEntry.match(/([^\)]\d{4})\):/)[1].replace('(', '');
```
Comment (jriedl) [inline ubtue_Brill.js:109]: How about
```javascript
  let dateCandidate = dateEntry.match(/\(\s*(\d{4})\s*\)/);
  return dateCandidate.length > 1 ? dateCandidate[1] : null;
```

Comment (socheres) [reply ubtue_Brill.js:109]: yes, your suggestion is better approach. Your regex will match also the second occurence, if the input text is like so "Archiv für katholisches Kirchenrecht - Volume 186 (2019): Issue 2 (2021)". Therefore i will change it to `/\(\s*(\d{4})\s*\)/:?` to match pattern followed by colon`:`

**Pull request merged**